src/modules/mavlink/mavlink_receiver.h

```diff
@@ -251,6 +252,8 @@ class MavlinkReceiver : public ModuleParams
 		uint8_t last_sequence{0};
 	};
 	ComponentState _component_states[MAX_REMOTE_COMPONENTS] {};
+	uint8_t _component_states_count{0};
+	static_assert((1 << sizeof(_component_states_count) * 8) > MAX_REMOTE_COMPONENTS, "MAVLink inappropriate counter type");
```

Comment:
Is this an excessive?

Reply:
What about just using an `unsigned` for `_component_states_count`?

Reply:
Of course `unsigned` will work, but why not save memory where it can be saved? While assertion is worth nothing.
I quite quite doubt that someone ever set MAX_REMOTE_COMPONENTS more than 256, especially knowing that sysid and compid max is 256 each. 

In fact assertion can be just removed with remaining `uint8_t` for `_component_states_count`, because `MAX_REMOTE_COMPONENTS  > 256` is such impracticable case, so user who wants this will know what he is doing.

Reply:
Alternatively I can make it `uint16_t ` - the fundamentally possible maximum number of external components.

Reply:
Yes, it's correct that there won't ever be more than 256.

In general, we use `unsigned` for all simple counter cases. I believe it's a trade off between RAM usage (as you bring up) and flash and CPU usage as using a `uint8_t` presumably requires a masking instruction. I haven't verified that though, and I don't know what the compiler optimizes away anyway.

Reply:
Ok, I've made it `unsigned`.